src/nodes.coffee

```diff
@@ -299,6 +313,12 @@ exports.Base = class Base
   astLocationData: ->
     jisonLocationDataToAstLocationData @locationData
 
+  # Mark AST nodes that correspond to expressions that (implicitly) return.
+  astAddReturns: (ast) ->
+    return ast unless ast?
```

Comment:
When is `ast` falsy?

Reply:
Eg `Elision` and `PassthroughLiteral` can both return `null` for their AST